### PR TITLE
Add awscli-local recipe

### PR DESCRIPTION
**Pull request by mwilson8**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by mwilson8** [comment]
@scopatz this package is ready for review

**Comment by mwilson8** [comment]
@oblute @benhuff will you be maintainers on this package?

File: recipes/awscli-local/meta.yaml~ [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{% set name = "awscli-local" %}
```
Comment (scopatz) [inline recipes/awscli-local/meta.yaml~]: Please remove this tilde file

File: recipes/awscli-local/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    - pip
+    - python
+  run:
+    - awscli
+    - localstack-client
+    - python
+
+test:
+  commands:
+    - aws help > /dev/null
```
Comment (scopatz) [inline recipes/awscli-local/meta.yaml]: Is there a real test for this?

Comment (mwilson8) [reply recipes/awscli-local/meta.yaml]: @scopatz May be able to execute a more robust command after "localstack" package has been added to the forge, but having a hard time finding an import style test for this package. All docs defer to aws package which is already in the forge.

Comment (mwilson8) [reply recipes/awscli-local/meta.yaml]: @scopatz awslocal.exe is also not recognized

The issues page of awscli-local seems to have others with this same problem: https://github.com/localstack/awscli-local/issues/6
https://github.com/localstack/awscli-local/issues/11

**Comment by mwilson8** [comment]
@scopatz The only other thing I can think to do is to just skip testing on windows; I can't find where the executable is installed (see multiple attempts: https://github.com/mwilson8/staged-recipes/commits/awslocal-test/recipes)

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**